spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -72,31 +73,34 @@ private SpannerStatementQueryExecutor() {
 	 * Executes a PartTree-based query.
 	 * @param type the type of the underlying entity
 	 * @param tree the parsed metadata of the query
-	 * @param params the parameters of this specific query
+	 * @param parameterAccessor the parameters of this specific query
 	 * @param queryMethodParamsMetadata parameter metadata from Query Method
 	 * @param spannerTemplate used to execute the query
 	 * @param spannerMappingContext used to get metadata about the entity type
 	 * @param <T> the type of the underlying entity
 	 * @return list of entities.
 	 */
-	public static <T> List<T> executeQuery(Class<T> type, PartTree tree, Object[] params,
+	public static <T> List<T> executeQuery(Class<T> type, PartTree tree, ParameterAccessor parameterAccessor,
 			Parameter[] queryMethodParamsMetadata,
 			SpannerTemplate spannerTemplate,
 			SpannerMappingContext spannerMappingContext) {
-		Pair<String, List<String>> sqlAndTags = buildPartTreeSqlString(tree,
-				spannerMappingContext, type);
+		SqlAndTags sqlAndTags = buildPartTreeSqlString(tree, spannerMappingContext, type, parameterAccessor);
 		Map<String, Parameter> paramMetadataMap = preparePartTreeSqlTagParameterMap(queryMethodParamsMetadata,
 				sqlAndTags);
+		Object[] params = StreamSupport.stream(parameterAccessor.spliterator(), false).toArray();
 		return spannerTemplate.query(type, buildStatementFromSqlWithArgs(
-				sqlAndTags.getFirst(), sqlAndTags.getSecond(), null,
+				sqlAndTags.getSql(), sqlAndTags.getTags(), null,
 				spannerTemplate.getSpannerEntityProcessor().getWriteConverter(), params, paramMetadataMap), null);
 	}
 
 	private static Map<String, Parameter> preparePartTreeSqlTagParameterMap(Parameter[] queryMethodParamsMetadata,
-			Pair<String, List<String>> sqlAndTags) {
+			SqlAndTags sqlAndTags) {
```

Comment:
I think a comment is necessary to explain why the params are filtered by `queryMethodParamsMetadatum.getType() != Pageable.class && queryMethodParamsMetadatum.getType() != Sort.class`